addons/project/wizard/project_task_share_wizard.py

```diff
@@ -1,10 +1,20 @@
-from odoo import fields, models
+from odoo import fields, models, api
 
 
 class TaskShareWizard(models.TransientModel):
     _name = 'task.share.wizard'
     _inherit = ['portal.share']
     _description = 'Task Sharing'
 
-    task_id = fields.Many2one('project.task', default=lambda self: self.res_id)
+    task_id = fields.Many2one('project.task', compute="_compute_task_id", store="true")
     project_privacy_visibility = fields.Selection(related='task_id.project_privacy_visibility')
+
+    @api.depends_context('active_id', 'active_model')
+    def _compute_task_id(self):
+        for rec in self:
+            ctx = rec.env.context
+            rec.task_id = (
+                ctx.get('active_id')
+                if ctx.get('active_model') == 'project.task'
+                else False
+            )
```

Comment:
We should set the active_id on the task instead of this. Please check above approach